Mathlib/RingTheory/AlgebraicIndependent/Transcendental.lean

```diff
@@ -70,30 +73,60 @@ theorem isEmpty_of_isAlgebraic [Algebra.IsAlgebraic R A] : IsEmpty ι := by
 
 end AlgebraicIndependent
 
+theorem trdeg_eq_zero_of_isAlgebraic [Algebra.IsAlgebraic R A] : trdeg R A = 0 :=
+  bot_unique <| ciSup_le' fun s ↦ have := s.2.isEmpty_of_isAlgebraic; (Cardinal.mk_eq_zero _).le
+
 open AlgebraicIndependent
 
 theorem AlgebraicIndependent.option_iff (hx : AlgebraicIndependent R x) (a : A) :
-    (AlgebraicIndependent R fun o : Option ι => o.elim a x) ↔
-      Transcendental (adjoin R (Set.range x)) a := by
+    AlgebraicIndependent R (fun o : Option ι ↦ o.elim a x) ↔
+      Transcendental (adjoin R (range x)) a := by
   rw [algebraicIndependent_iff_injective_aeval, transcendental_iff_injective,
     ← AlgHom.coe_toRingHom, ← hx.aeval_comp_mvPolynomialOptionEquivPolynomialAdjoin,
     RingHom.coe_comp]
   exact Injective.of_comp_iff' (Polynomial.aeval a)
     (mvPolynomialOptionEquivPolynomialAdjoin hx).bijective
 
-/-- Variant of `algebraicIndependent_of_finite_type` using `Transcendental`. -/
-theorem algebraicIndependent_of_finite_type'
-    (hinj : Injective (algebraMap R A))
+theorem AlgebraicIndependent.optionElim_iff {a : A} :
```

Comment:
The name of this theorem is a little bit confusing, probably because the previous `AlgebraicIndependent.option_iff` is also. Both of them mentions `Option.elim`, and seems that the existing `AlgebraicIndependent.option_iff` is just a special case of `AlgebraicIndependent.optionElim_iff`. Do you think we should called the existing one `option_iff_transcendental`, and the newly added one `option_iff`?

Reply:
Okay, let me do what you said.